### DISCUSSION TITLE
How to add additional command or additional package in k3os?

### DISCUSSION THREAD

**Discussion by kuddusrhce** [rancher/k3os#798] [Q&A]
Hello Team,

We have created K3OS OS images from the packer and created a virtual machine and hosted them in a virtual environment. We are looking for a password reset option in the existing machine for rancher user to force to change password at first login. We haven't found any change and passwd command. Also, we have realized that passwd is linked to busybox command. It doesn't have passwd -e option? How to proceed further? Is there any way to add/install additional packages so that we can use either chage or passwd -e option? Please suggest.

**Comment by dweomer** [comment]
k3OS is not meant to be a traditionally multi-user operating system. The `rancher` user is all that is supported, the password for which is sourced from `config.yaml` at every boot. That said, I do not recommend the use of passwords but instead add ssh public keys to the `rancher` user profile (also supported via `config,yaml`).

**Comment by kuddusrhce** [comment]
Okay got it. Also Is there any way to add command or any package in existing k3os? If yes then please guide me.

**Comment by dweomer** [comment]
> Okay got it. Also Is there any way to add command or any package in existing k3os? If yes then please guide me.

You can either statically compile it (recommended) or build it with dynamic linking in an alpine 3.12 container and copy it (along with any dependent libs not already included in the base install) over to your k3OS system(s). Alternatively, consider a custom build of k3OS (see [images/00-base/Dockerfile](https://github.com/rancher/k3os/blob/master/images/00-base/Dockerfile)).

If you are looking to install a package the you will have to copy the binary and any other support files onto your k3OS system(s). What this looks like depends on the package.
